msteams-platform/m365-apps/build-actions-in-m365.md

```diff
@@ -0,0 +1,165 @@
+| &nbsp; | [Microsoft Teams](https://www.microsoft.com/microsoft-teams/download-app) | Microsoft Teams to collaborate with everyone you work with through apps for chats, meetings, and calls in one place.|
+| &nbsp; | [Node.js](https://nodejs.org/en/download/) | Back-end JavaScript runtime environment. For more information, see [Node.js version compatibility table for project type](~/toolkit/build-environments.md#nodejs-version-compatibility-table-for-project-type).|
+| &nbsp; | [Microsoft Edge](https://www.microsoft.com/edge) (recommended) or [Google Chrome](https://www.google.com/chrome/) | A browser with developer tools. |
+| &nbsp; | [Visual Studio Code](https://code.visualstudio.com/download) | JavaScript, TypeScript, or SharePoint Framework (SPFx) build environments. Use the latest version. |
+
+To build Actions for your app, follow these steps:
+
+> [!div class="checklist"]
+>
+>* [Update app manifest](#update-app-manifest).
+>* [Retrieve Action information through context object](#retrieve-action-information-through-context-object).
+>* [Access content through Graph API](#access-content-through-graph-api).
+
+## Update app manifest
+
+Define the intent, object, and handler for your actions in the app manifest (previously called Teams app manifest).
+
+The following is an example of the `intent`, `supportedobjects`, and `handlers` properties for `openPage` in app manifest:
+
+```json
+{
+  "$schema": "https://developer.microsoft.com/json-schemas/teams/vDevPreview/MicrosoftTeams.schema.json",
+  "manifestVersion": "devPreview",
+.
+.
+.
+"actions": [
+    {
+        // Defining Action to open a page
+        "id": "relatedTasks",
+        "displayName": "Related tasks",
+        "intent": "custom",
+        "description": "Shows tasks in the To do app that are related to this file.",
+        "handlers": [
+            {
+                "type": "openPage",
+                "supportedObjects": {
+                    "file": {
+                        "extensions": ["xlsx", "doc", "docx", "pdf", "pptx", "ppt"]
+                    }
+                },
+                "pageInfo": {
+                    "pageId": "index",
+                }
+            }
+        ]
+    }
+]
+}
+```
+
+For more information, see [public developer preview app manifest schema](../resources/schema/manifest-schema-dev-preview.md#actions).
+
+When a user selects an action to open a page and view related tasks in an app based on the selected file. The app uses the `"intent": "custom"` property to identify the file type, such as .xlsx or doc and the `"type": "openPage"` handler opens the app and navigates to the `pageId`.
+
+## Retrieve Action information through context object
+
+Update the handler to receive the Action information through the context object to create a seamless user experience for performing users specific tasks using the [Teams JavaScript library (TeamsJS)](/javascript/api/@microsoft/teams-js).
```

Comment:
User need to build the handler first. So instead of using "update" you can say "Build the handler, in this case the personal tab, to receive ..."

Reply:
2. Please add link to the context object. 

https://learn.microsoft.com/en-us/javascript/api/%40microsoft/teams-js/app.context?view=msteams-client-js-latest